frontend/public/module/k8s/k8s-models.ts

```diff
@@ -77,6 +82,39 @@ export const modelFor = (ref: K8sResourceKindReference) => {
   }
 };
 
+/**
+ * Provides a synchronous way to acquire an API discovered Kubernetes model by group and kind only.
+ * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
+ */
+export const modelForGroupKind = (group: string, kind: string): K8sKind => {
+  const models: ImmutableMap<string, K8sKind> = store.getState().k8s.getIn(['RESOURCES', 'models']);
```

Comment:
We want to avoid using the global redux store.

cc @vojtechszocs

Reply:
well Vojtech already approved this change so it got in inside https://github.com/openshift/console/pull/9355
would like to hear if there's another solution that is viable for both React and non-React context

Reply:
If we need data from application's Redux store outside a React component, it should be OK to use `store.getState()`.

I think @christianvogt is referring to the `import store` declaration which adds dependency on `public/redux.ts` module, we can refactor the code to avoid that static import if needed.

Reply:
the `import store` has been inside this file for over 4 years :)

Reply:
No my comment was about the new use of `store.getState()`.
I don't want to see more use cases that use the global store because it's bad practice.
Yes the import has been around a while now because removing these instances is difficult.

Reply:
> If we need data from application's Redux store outside a React component, it should be OK to use store.getState()

@christianvogt I see no other way here when dealing with contexts that are out of React.